compiler/plugins/target/AMD-AIE/aievec/VectorToVectorConversions.cpp

```diff
@@ -39,6 +40,73 @@ namespace mlir::iree_compiler::aievec {
 
 using namespace mlir;
 
+struct CanonicalizeTrivialReadAccessSubviewOpPattern
```

Comment:
Canonicalizations in general are supposed to be implemented with rewrite pattern.

Let me know if I understood "standard rewrite function" correctly : do you mean `funcName(IRRewriter &rewriter, vector::TransferReadOp readOp)` ?

One case where that function wouldn't work is the following case :-
```
%b = subview %a
%c = subview %b
%d = transfer_read %c
```
This needs to go through stages :-
```
%b = subview %a
%d = transfer_read %b
```
and then :-
```
%d = transfer_read %a
```

These and mostly other things are better off handled via a Rewrite pattern which runs in a loop till no change in the DAG takes place.

Reply:
What is the initial motivation for this pattern, why can't lowering handle the IR with a subview followed by a transfer_read, and will we still fall over if the transfer_read is not "trivial" (all zero offsets) ? 

It would be useful to have a short summary in the codebase IMO (apologies if it's already there, I haven't looked at the full PR yet).